src/network/network_udp.cpp

```diff
@@ -646,7 +646,10 @@ void NetworkUDPClose()
 /** Receive the UDP packets. */
 void NetworkBackgroundUDPLoop()
 {
-	std::lock_guard<std::mutex> lock(_network_udp_mutex);
+	/* Try to take the lock, but do nothing if the mutex is locked by another thread.
+	 * Avoid blocking the entire main network loop over handling incoming UDP in a timely manner. */
+	if (!_network_udp_mutex.try_lock()) return;
+	std::lock_guard<std::mutex> lock(_network_udp_mutex, std::adopt_lock);
```

Comment:
I can be wrong, but I believe the atomic way for this is:

```
std::unique_lock<std::mutex> lock(_network_udp_mutex, std::adopt_lock);
if (!lock.try_lock()) return;
```

But I am sure someone else will tell me wrong :P Either way, your approach is not atomic, so from a threading perspective it could still hang, I think?

Reply:
I think they are equivalent? If the first line in mine takes the lock, the lock is held. The lock_guard then adopts the lock and will release it on return. If taking the lock failed, then the function returns without the lock_guard being constructed.

Reply:
> But I am sure someone else will tell me wrong
It is wrong.

If the lock is already locked, try_lock will return false and the unique lock will try to free the lock when the current thread does not own it, which causes undefined behaviour.

The code is correct but it would be less unclear if it used std::unique_lock::try_lock instead.